HotFix/Core/MessageWriter.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Text;
+using HotFix.Utilities;
+
+namespace HotFix.Core
+{
+    public class MessageWriter
+    {
+        private readonly byte[] _buffer;
+        private int _end;
+
+        private readonly int _headerEnd;
+        private int _bodyEnd;
+        private int _trailerEnd;
+
+
+        public MessageWriter(int maxLength, string beginString)
+        {
+            _buffer = new byte[maxLength];
+
+            _end += _buffer.WriteString(_end, "8=");
+            _end += _buffer.WriteString(_end, beginString);
+            _buffer[_end++] = (byte)'\u0001';
+
+            _end += _buffer.WriteString(_end, "9=");
+            _end += _buffer.WriteString(_end, "00000");
+            _buffer[_end] = (byte)'\u0001';
+
+            _headerEnd = _end;
+            _end++;
+
+            _end += _buffer.WriteString(_end, "35=");
+        }
+
+        public void Prepare(string messageType)
+        {
+            _end = _headerEnd + 4;
+
+            _end += _buffer.WriteString(_end, messageType);
+            _buffer[_end++] = (byte)'\u0001';
+        }
+
+        public void Set(int tag, string value)
+        {
+            _end += _buffer.WriteInt(_end, tag);
+            _buffer[_end++] = (byte) '=';
+
+            _end += _buffer.WriteString(_end, value);
+            _buffer[_end++] = (byte)'\u0001';
```

Comment:
It might be easy to break this by accidentally mistyping/erasing a part of the `equals` and `soh` chars so we should extract them to consts, ie:

``` csharp
private const char EQL = '=';
private const char SOH = '\u0001';
```